test/e2e/services.go

```diff
@@ -0,0 +1,378 @@
+// +build integration interactive
+
+package e2e
+
+import (
+	"os"
+	"path/filepath"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/efficientgo/e2e"
+	"github.com/efficientgo/tools/core/pkg/testutil"
+)
+
+const (
+	apiImage = "quay.io/observatorium/api:latest"
+
+	thanosImage = "quay.io/thanos/thanos:v0.22.0"
+	lokiImage   = "grafana/loki:2.3.0"
+	upImage     = "quay.io/observatorium/up:master-2021-02-12-03ef2f2"
+
+	dexImage        = "dexidp/dex:v2.30.0"
+	opaImage        = "openpolicyagent/opa:0.31.0"
+	gubernatorImage = "thrawn01/gubernator:1.0.0-rc.8"
+
+	logLevelError = "error"
+	logLevelDebug = "debug"
+)
+
+func startServicesForMetrics(t *testing.T, e e2e.Environment) (
+	metricsReadEndpoint string,
+	metricsWriteEndpoint string,
+	metricsExtReadEndppoint string,
+) {
+	thanosReceive := newThanosReceiveService(e)
+	thanosQuery := newThanosQueryService(e, thanosReceive.InternalEndpoint("grpc"))
+	testutil.Ok(t, e2e.StartAndWaitReady(thanosReceive, thanosQuery))
+
+	return thanosQuery.InternalEndpoint("http"),
+		thanosReceive.InternalEndpoint("remote_write"),
+		thanosQuery.Endpoint("http")
+}
+
+func startServicesForLogs(t *testing.T, e e2e.Environment) (
+	logsEndpoint string,
+	logsExtEndpoint string,
+) {
+	loki := newLokiService(e)
+	testutil.Ok(t, e2e.StartAndWaitReady(loki))
+
+	return loki.InternalEndpoint("http"),
+		loki.Endpoint("http")
```

Comment:
Is it just me or do these indents look like double tabs? They seem way bigger than normal :sweat_smile:

Reply:
Not sure :shrug: Either way that newline here is redundant.